src/Servers/Kestrel/Core/test/ChunkWriterTests.cs

```diff
@@ -43,5 +43,21 @@ public void ChunkedPrefixMustBeHexCrLfWithoutLeadingZeros(int dataCount, string
 
             Assert.Equal(Encoding.ASCII.GetBytes(expected), span.Slice(0, count).ToArray());
         }
+
+        [Fact]
+        public void ChunkedPrefixReturnsSegmentThatDoesNotNeedToMove()
```

Comment:
How long does this test take? It doesn't cover some edge cases like i = 268435467 since it doesn't loop that high.

Instead of testing every i from 5 to 10000000 (5 isn't even valid), it might be better to just test the boundary cases. So basically 6, int.MaxVales, the constants used in GetPrefixBytesForChunk, and the values neighboring those constants.

Reply:
The only reason I didn't test 268435467 was due to OOM. It doesn't take long (less than a second).

Reply:
I'll rewrite the test to not allocate a large array.

Reply:
Bleh I don't like how I would need to rewrite it. I would need to either change GetPrefixBytesForChunk to pass in a length or do something hacky.